### ISSUE TITLE
fromDateTime() return type differs between V1 vs V2

### ISSUE THREAD

**Issue by ksnijjer** [apache/pinot#12612]
Discussed this with @Jackie-Jiang , currently in V1 above function returns a long which means an expression like this
DATE_MINUTE_EPOCH > fromDATETIME('2023-02-01 00:00:00.000','yyyy-MM-dd HH:mm:ss.SSS')

where first col. type is Long (epochMillis) works without any issue. However in V2 return type for the function is Timestamp as such above expression has to be explicitly cast as BIGINT to make it work, i.e
DATE_MINUTE_EPOCH > cast(fromDATETIME('2023-02-01 00:00:00.000','yyyy-MM-dd HH:mm:ss.SSS') as BIGINT)
else it fails with a type cast error.

Filing this issue as a placeholder to discuss how should this be handled in V2 vs V1 behavior given the data type definitions are quite different.

**Comment by GopsAB** [comment]
@Jackie-Jiang Do you need help to analyse and fix it if it is an issue?

**Comment by Jackie-Jiang** [comment]
@GopsAB Yes. The root cause is that a lot of date time functions can only take TIMESTAMP as input. In V1, `TIMESTAMP` and `LONG` can be used interchangeable because the internal storage are the same. It is not the same case in V2 which is strong typed. Using `LONG` column as input can cause error during planning. IMO the correct fix should be to make sure all functions are SQL compatible, i.e. the return value matches the standard SQL definition

**Comment by yashmayya** [comment]
`fromDateTime` doesn't seem to be a standard SQL function that exists in other databases. I think it probably makes sense to unify the v2 behavior with v1 in this case - i.e., it should take a timestamp string as input and return a long representing milliseconds since epoch.
